Resources/Private/JavaScript/DlfMediaPlayer/components/waveform/WaveForm.js

```diff
@@ -0,0 +1,392 @@
+      },
+      waveformData: {
+        arraybuffer: waveformdata.toArrayBuffer(),
+      },
+      waveformCache: true,
+    };
+
+    Peaks.init(options, (err, peaks) => {
+      if (!peaks) {
+        console.log(err);
+        return;
+      }
+
+      // The player may hide poster on manual seek
+      // TODO: Find a more robust solution
+      peaks.player.seek = (time) => {
+        this.player?.seekTo(time);
+      };
+
+      this.peaks_ = peaks;
+
+      this.zoomview = peaks.views.getView('zoomview');
+      this.zoomview?.setZoom({
+        seconds: waveformdata.duration,
+      });
+      this.zoomview?.setWheelMode('scroll');
+
+      this.adaptPeaksMarkers();
+    });
+  }
+
+  /**
+   * @private
+   */
+  adaptPeaksMarkers() {
+    if (this.peaks_ === null || this.player === null) {
+      return;
+    }
+
+    // TODO: Also call "off", and handle new Peaks instance ("waveformHandlersRegistered")?
+
+    this.peaksAddSegments(this.player.getMarkers().getSegments());
+
+    for (const [event, handler] of Object.entries(this.peaksHandlers)) {
+      // @ts-expect-error: `Object.entries()` is too coarse-grained
+      this.peaks_.on(event, handler);
+    }
+
+    for (const [event, handler] of Object.entries(this.markersHandlers)) {
+      this.player.getMarkers()
+        // @ts-expect-error: `Object.entries()` is too coarse-grained
+        .addEventListener(event, handler);
+    }
+  }
+
+  /**
+   * @private
+   * @param {import('DlfMediaPlayer/Markers').Segment[]} segments
+   */
+  peaksAddSegments(segments) {
+    if (this.peaks_ !== null) {
+      for (const segment of segments) {
+        if (segment.endTime === undefined) {
+          this.peaks_.points.add({
+            id: segment.id,
+            time: segment.startTime,
+            labelText: segment.labelText,
+            color: segment.color,
+            editable: segment.editable,
+          });
+        } else {
+          this.peaks_.segments.add({
+            id: segment.id,
+            startTime: segment.startTime,
+            endTime: segment.endTime,
+            labelText: segment.labelText,
+            color: segment.color,
+            editable: segment.editable,
+          });
+        }
+      }
+    }
+  }
+
+  /**
+   * @private
+   * @param {WheelEvent} e
+   */
+  onWheel(e) {
+    const zoomInFactor = 1.5 ** (-e.deltaY / 100);
+    this.updateZoom(this.samplesPerPixel / zoomInFactor);
+  }
+
+  /**
+   * @private
+   */
+  onResize() {
+    this.zoomview?.fitToContainer();
+    this.updateZoom();
+  }
+
+  /**
+   * @private
+   * @param {number} samplesPerPixel
+   */
+  updateZoom(samplesPerPixel = this.samplesPerPixel) {
+    if (this.waveformdata === null) {
+      return;
+    }
+
+    const { sample_rate, seconds_per_pixel, duration } = this.waveformdata;
+
+    this.minSamplesPerPixel = seconds_per_pixel * sample_rate;
+    this.maxSamplesPerPixel = clamp(
+      duration * sample_rate / this.$waveOverview.offsetWidth,
```

Comment:
```suggestion
    const { sampleRate, secondsPerPixel, duration } = this.waveformdata;

    this.minSamplesPerPixel = secondsPerPixel * sampleRate;
    this.maxSamplesPerPixel = clamp(
      duration * sampleRate / this.$waveOverview.offsetWidth,
```
Please use `camelCase` for variables and properties.

Reply:
f5243b944ef1f1b2b206e599b6e25050377f503a
revert variable renaming as it would break compatibility with waveform-data library